src/App.tsx

```diff
@@ -15,71 +16,93 @@ export const goods = [
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  handleSelect = (good: string) => {
+    this.setState({ selectedGood: good });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good">
-          <td>
+    return (
+      <main className="section container">
+        <h1
+          className={`title ${
+            selectedGood
+              ? 'is-flex is-align-items-center'
+              : ''
+          }`}
+        >
+          {selectedGood
+            ? `${selectedGood} is selected`
+            : 'No goods selected'}
+          {selectedGood && (
+            // eslint-disable-next-line jsx-a11y/control-has-associated-label
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={this.handleClear}
+            />
+          )}
+        </h1>
+        <table className="table">
+          <tbody>
+            {goods.map((good) => {
+              const isSelected = selectedGood === good;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+              return (
+                <tr
+                  key={good}
+                  data-cy="Good"
+                  className={classnames({
+                    'has-background-success-light': isSelected,
+                  })}
+                >
+                  <td>
+                    {isSelected
+                      ? (
+                        <button
+                          data-cy="RemoveButton"
+                          type="button"
+                          className="button is-info"
+                          onClick={this.handleClear}
+                        >
+                          -
+                        </button>
+                      )
+                      : (
+                        <button
+                          data-cy="AddButton"
+                          type="button"
+                          className="button"
+                          onClick={() => this.handleSelect(good)}
+                        >
+                          +
+                        </button>
+                      )}
```

Comment:
As for me this syntax is better but it's up to you
```suggestion
                    {isSelected ? (
                      <button 
                        data-cy="RemoveButton" 
                        type="button" 
                        className="button is-info" 
                        onClick={this.handleClear}
                      >
                        -
                      </button>
                    ) : (
                      <button
                        data-cy="AddButton" 
                        type="button" 
                        className="button" 
                        onClick={() => this.handleSelect(good)}
                      >
                        +
                      </button>
                    )}
```